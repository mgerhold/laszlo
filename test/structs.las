struct S {
    name: String,
    age: I32,
}

let s = new S {
    name: "Claus Kleber",
    age: 67,
};

let s2 = new S {
    name: "Gundula Gause",
    age: 58,
};

let persons = [
        new S {
            name: "Claus Kleber",
            age: 67,
        },
        new S {
            name: "Gundula Gause",
            age: 58,
        }];

println(typeof(s));
println(s);
println(s == s);
println(s == s2);
println(s.name);
s.name = "Claus-Detlev Walter Kleber";
println(s);

persons[0].name = "Claus-Detlev Walter Kleber";
println(persons[0]);
