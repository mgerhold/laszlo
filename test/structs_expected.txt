struct S(name: String, age: I32)
Struct S(name: Claus Kleber, age: 67)
true
false
Claus Kleber
Struct S(name: Claus-Detlev Walter Kleber, age: 67)